activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/MessageProducerActivitiEventListener.java

```diff
@@ -24,10 +29,34 @@ public MessageProducerActivitiEventListener(ProcessEngineChannels producer,
 
     @Override
     public void onEvent(ActivitiEvent event) {
+        CommandContext currentCommandContext = Context.getCommandContext();
         ProcessEngineEvent newEvent = converterContext.from(event);
-        if (newEvent != null) {
-            producer.auditProducer().send(MessageBuilder.withPayload(newEvent).build());
+        if (newEvent == null) {
+            return;
+        }
+        @SuppressWarnings("unchecked")
+        List<ProcessEngineEvent> events = (List<ProcessEngineEvent>) currentCommandContext
```

Comment:
I was wordering if we cannot move this `@SuppressWarnings("unchecked")` and cast to `CommandContext` class using generics.
- add the following method to `CommandContext`
```
@SuppressWarnings("unchecked")
  public <T> T getGenericAttribute(String key) {
    return (T) getAttribute(key);
  }
```
- Update the line above to
```
List<ProcessEngineEvent> events = currentCommandContext.getGenericAttribute(MessageProducerCommandContextCloseListener.PROCESS_ENGINE_EVENTS);
```
The advantage is that we do not need to add `@SuppressWarnings("unchecked")` neither cast on potential new listeners (i.e in `MessageProducerCommandContextCloseListener `). What do you think?